### ISSUE TITLE
macOS accessibility: Releasing ctrl after opening ctrl+tab quick switch menu does not dismiss it, when using Keyboard Viewer or Synergy

### ISSUE THREAD

**Issue by kibiz0r** [microsoft/vscode#144801]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.65.1 (Universal)
- OS Version: macOS 12.2.1 (21D62)

Steps to Reproduce:

My personal complaint here is when using VSCode through Synergy, but the same issue is present when using macOS's own built-in Keyboard Viewer, so this looks like an accessibility bug beyond an arbitrary side-effect of some third-party software.

Keyboard Viewer steps:

1. In VSCode for Mac, open multiple files
2. Open the [Keyboard Viewer](https://support.apple.com/guide/mac-help/use-the-keyboard-viewer-on-mac-mchlp1015/mac) accessibility tool
3. Click the ctrl key
4. Click the tab key
5. See that the VSCode quick switch menu opens
6. See that the keyboard state no longer has the ctrl key held
7. **See that the VSCode quick switch menu is still open** <-- problem
8. (The menu can still be dismissed/accepted by clicking the ctrl key on and off again, or with the enter key)

Synergy steps:

1. Get a Mac and a Windows PC on the same network
9. Install VSCode and [Synergy](https://symless.com/synergy) (1.14.3) on the Mac
10. Install [Synergy](https://symless.com/synergy) (1.14.3) on the PC
11. Configure Synergy to send the PC mouse+keyboard input to the Mac
12. In VSCode on the Mac, open multiple files
13. Verifying behavior of non-Synergy UX:
14. Using a keyboard **directly attached to the Mac**...
15. ...**hold** ctrl
16. ...**press** tab
17. ...see that the VSCode quick switch menu opens
18. ...**release** ctrl
19. ...see that the VSCode quick switch menu closes
20. Demonstrating buggy behavior of Synergy UX:
21. Using **Synergy** to send keyboard input **from the PC to the Mac** over the network...
22. ...**hold** ctrl
23. ...**press** tab
24. ...see that the VSCode quick switch menu opens
25. ...**release** ctrl
26. ...**see that the VSCode quick switch menu is still open!** <-- problem
27. (The menu can still be dismissed/accepted with the enter key)

You can further verify that both Keyboard Viewer and Synergy seem to be maintaining the correct keyboard state by using KeyCastr on the Mac (with KeyCastr's "Svelte" visualizer selected). KeyCastr shows that the ctrl key is released, but VSCode still keeps the quick switch menu open.

**Comment by TylerLeonhardt** [comment]
This is interesting... have you experienced this behavior with other keybindings?

@alexdima is there a chance that how we handle keybindings could be causing this?

**Comment by alexdima** [comment]
The quick pick is independently listening for a `keyup` event [here](https://github.com/microsoft/vscode/blob/3e5c7e2c570a729e664253baceaf443b69e82da6/src/vs/base/parts/quickinput/browser/quickInput.ts#L895) outside of the keybinding dispatching, because the keybinding service dispatches on `keydown`. That code might have a problem, or the keyboard viewer might not be sending keyboard events in 100% the same way as the physical keyboard.

**Comment by kibiz0r** [comment]
> have you experienced this behavior with other keybindings?

No, I haven't. I wouldn't call my testing *exhaustive* here, by any means, but I have put a few dozen hours into this setup and it seems to work fine except for ctrl+tab.

**Comment by afriza** [comment]
I found this issue thread from [Synergy's support](https://symless.com/synergy/contact-support) team.

My issue is different but also related to keypress.

The [GitLens extension](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) for VS Code [doesn't capture](https://github.com/gitkraken/vscode-gitlens/issues/2385) <kbd>⌥ opt</kbd> + _click_ combination and only detects the _click_ without <kbd>⌥ opt</kbd>.

I thought this is an issue in Synergy but [KeyCastr](https://github.com/keycastr/keycastr) is actually able to show both <kbd>⌥ opt</kbd> and _click_.

So, the problem is probably in VS Code and/or GitLens extension.

**Comment by kibiz0r** [comment]
@TylerLeonhardt Any thoughts on prioritizing this? Synergy may be a less-common way to use VSCode, but for the people who do use VSCode through Synergy, it's an all-day thing and it's hard to ignore the friction caused by an awkward ctrl-tab experience.
